src/core/Akka/Util/ByteString.cs

```diff
@@ -519,16 +427,14 @@ public ByteString Concat(ByteString other)
             if (other == null) throw new ArgumentNullException(nameof(other), "Cannot append null to ByteString.");
 
             if (other.IsEmpty) return this;
-            if (this.IsEmpty) return other;
+            if (IsEmpty) return other;
 
-            var count = _count + other._count;
-            var len1 = _buffers.Length;
-            var len2 = other._buffers.Length;
-            var array = new ByteBuffer[len1 + len2];
-            Array.Copy(this._buffers, 0, array, 0, len1);
-            Array.Copy(other._buffers, 0, array, len1, len2);
-
-            return new ByteString(array, count);
+            // combine the two ReadOnlyMemory<byte> instances
+            var array = new byte[_memory.Length + other._memory.Length];
```

Comment:
possible over-opt:
not sure whether compiler will wind up reading length twice here?

Reply:
yeah I guess I could use the local variables but I think this is probably fine